scripts/lib.sh

```diff
@@ -213,3 +213,96 @@ EOT
 	local rootfs_file="${file_dir}/$(basename "${file}")"
 	info "Created summary file '${rootfs_file}' inside rootfs"
 }
+
+# generate_dockerfile takes as only argument a path. It expects a Dockerfile.in
+# Dockerfile template to be present in that path, and will generate a usable
+# Dockerfile replacing the '@PLACEHOLDER@' in that Dockerfile
+generate_dockerfile()
+{
+	dir="$1"
+	[ -d "${dir}" ] || die "${dir}: not a directory"
+
+	case "$(uname -m)" in
+		"ppc64le")
+			goarch=ppc64le
+			;;
+
+		"aarch64")
+			goarch=arm64
+			;;
+		"s390x")
+			goarch=s390x
+			;;
+
+		*)
+			goarch=amd64
+			;;
+	esac
+
+	[ -n "${http_proxy:-}" ] && readonly set_proxy="RUN sed -i '$ a proxy="${http_proxy:-}"' /etc/dnf/dnf.conf /etc/yum.conf; true"
+
+	curlOptions=("-OL")
+	[ -n "${http_proxy:-}" ] && curlOptions+=("-x ${http_proxy:-}")
+	readonly install_go="
+RUN cd /tmp ; curl ${curlOptions[@]} https://storage.googleapis.com/golang/go${GO_VERSION}.linux-${goarch}.tar.gz
+RUN tar -C /usr/ -xzf /tmp/go${GO_VERSION}.linux-${goarch}.tar.gz
```

Comment:
Nit: `/usr/local/`?

Reply:
[This commit](https://github.com/kata-containers/osbuilder/pull/312/commits/eb186f9de4d0ef487aa024f4f9af029c4826910f) just moved 2 functions from rootfs.sh to lib.sh. If `/usr/local` is preferred, it's worth creating a new issue.